intent-classifier/training/csv_to_tensors.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+/**
+ * Takes a CSV file of queries and intents and embeds all the queries using the
+ * universal sentence encoder and one-hot-encodes the queries.
+ * The output of this script will be an JSON file with the following format
+ *
+ *  {
+ *    xs: number[][], // Array of tensor-like objects for queries
+ *    ys: number[][], // Array of tensor-like objects for intents (labels)
+ *  }
+ *
+ * The script will also output a `metadata.json` file with the following format:
+ *
+ *  {
+ *    labels: string[], array of strings to invert one-hot encodings
+ *    xsShape: [number, number], tensor shape of xs
+ *    ysShape: [number, number], tensor shape of ys
+ *  }
+ */
+
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+
+const tf = require('@tensorflow/tfjs');
+const _ = require('lodash');
+const argparse = require('argparse');
+global.fetch = require('node-fetch');
+const useLoader = require('@tensorflow-models/universal-sentence-encoder');
+
+const {loadCSV} = require('./util');
+
+
+async function convertToTensors(data, labels, use, batchSize) {
+  // Only keep the queries for the intents we are interested in.
+  const filtered = data.filter(d => labels.indexOf(d.intent) >= 0);
+  console.log(`Converting ${filtered.length} queries to tensors.`);
+
+  // Shuffle the data before we tensorify it and write it out.
+  tf.util.shuffle(filtered);
+
+  const xs = filtered.map(d => d.query);
+  const ys = filtered.map(d => labels.indexOf(d.intent));
+
+  const xsBatches = _.chunk(xs, batchSize);
```

Comment:
can we avoid lodash, it does put another burden on the reader to familiarilize herself with another library.